src/compiler/checker.ts

```diff
@@ -26912,6 +26912,9 @@ namespace ts {
                 errorOutputContainer);
 
             function checkTagNameDoesNotExpectTooManyArguments(): boolean {
+                if (getJsxNamespaceContainerForImplicitImport(node)) {
+                    return true; // factory is implicitly jsx/jsxdev - assume it fits the bill, since we don't strongly look for the jsx/jsxs/jsxDEV factory APIs anywhere else (at least not yet)

```

Comment:
This makes sense, but it’s unintuitive as to why it fixes this particular bug. Something later in this function marked `React` as referenced, for some reason?

Reply:
Yeah - looking up it's type to see if it had call signatures.